lib/ts/logger.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import debug from "debug";
+import { version } from "./version";
+
+/*
+ The debug logger and info logger defined below use the debug lib to log debug and info messages when the DEBUG env is set with the com.supertokens namespace.
+For example:
+    Adding a info log to the emailpassword signin api
+    Adding a debug log to the SignIn API which will print the status when a non OK response is returned
+Flow: 
+    SignIn with with valid credentials;
+    SignIn with invalid credentials
+Output: (with DEBUG=com.supertokens:*)
+  com.supertokens:info {t: "1647340361090", msg: "SignInAPI is called", code: 2, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:54:69", sdkVer: "9.1.0"} +0ms
+  com.supertokens:debug {t: "1647340361295", msg: "SignInAPI replied with status: OK", code: 1, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:65:76", sdkVer: "9.1.0"} +0ms
+  com.supertokens:info {t: "1647340563421", msg: "SignInAPI is called", code: 2, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:54:69", sdkVer: "9.1.0"} +0ms
+  com.supertokens:debug {t: "1647340563565", msg: "SignInAPI replied with status: WRONG_CREDENTIALS_ERROR", code: 1, file: "/home/supertokens-node/lib/build/recipe/emailpassword/api/signin.js:71:76", sdkVer: "9.1.0"} +0ms
+*/
+
+export let loggerCodes = {
+    API_RESPONSE: 1,
+    API_CALLED: 2,
+};
+
+let logMessage = (namespaceId: string, message: string, code: number | undefined) => {
+    let fileNameAndLineNumber = getFileLocation();
+    let messageWithOptionalCode = `msg: \"${message}\"`;
+    if (code != undefined) {
+        messageWithOptionalCode = `${messageWithOptionalCode}, code: ${code}`;
+    }
+    debug(`com.supertokens:${namespaceId}`)(
+        `{t: "${Date.now()}", ${messageWithOptionalCode}, file: \"${fileNameAndLineNumber}\" sdkVer: "${version}"}`
+    );
+};
+
+export let infoLoggerWithCode = {
+    [loggerCodes.API_CALLED]: (apiName: string) => {
+        logMessage("info", `${apiName} is called`, loggerCodes.API_CALLED);
+    },
+};
+
+export let debugLoggerWithCode = {
+    [loggerCodes.API_RESPONSE]: (apiName: string, status: string) => {
```

Comment:
I think the call log should be debug-level and the response info since the response log includes more information and it's more likely to be useful on its own.

Reply:
Agreed